store/tikv/mock-tikv/mvcc.go

```diff
@@ -241,6 +241,40 @@ func (s *MvccStore) Scan(startKey, endKey []byte, limit int, startTS uint64) []P
 	return pairs
 }
 
+// Seek searches for the first key in the engine which is >= key in byte order, returns (nil, nil, ErrNotFound)
+// if such key is not found.
+func (s *MvccStore) Seek(key []byte, startTS uint64) Pair {
+	return s.doSeek(key, startTS, false)
+}
+
+// SeekReverse searches the engine in backward order for the first key-value pair which key is less than the key
+// in byte order, returns (nil, nil, ErrNotFound) if such key is not found. If key is nil, the last key is returned.
+func (s *MvccStore) SeekReverse(key []byte, startTS uint64) Pair {
+	return s.doSeek(key, startTS, true)
+}
+
+func (s *MvccStore) doSeek(key []byte, startTS uint64, reverse bool) Pair {
+	var pair Pair
+	iterator := func(item llrb.Item) bool {
+		k := item.(*mvccEntry).key
+		if reverse && bytes.Equal(k, key) {
+			return true
+		}
+		val, err := s.Get(k, startTS)
+		if val != nil || err != nil {
+			pair = Pair{Key: k, Value: val, Err: err}
+			return false
+		}
+		return true
+	}
+	if reverse {
+		s.tree.DescendLessOrEqual(newEntry(key), iterator)
+	} else {
+		s.tree.AscendGreaterOrEqual(newEntry(key), iterator)
+	}
```

Comment:
Access the tree without lock here. I suggest just using `scan 1` like tikv-server.